picamraw/main.py

```diff
@@ -244,7 +244,13 @@ def _unpack_10bit_values(pixel_bytes_2d):
 
     # In each row, split up every 5th byte and unpack it into the low 2-bits of the four preceding bytes
     for byte in range(4):
-        data[:, byte::5] |= ((data[:, 4::5] >> ((4 - byte) * 2)) & 3)
+        fifth_bytes = pixel_bytes_2d[:, 4::5]
+        # Shift the bits over so that the relevant ones are in the rightmost (lowest) position
+        # eg. for byte 1, 0b00001100 -> 0b11
+        shifted_bits_from_fifth_bytes = fifth_bytes >> (byte * 2)
+        # Mask the relevant ones (the lowest 2)
+        masked_low_bits = shifted_bits_from_fifth_bytes & 0b11
+        data[:, byte::5] |= masked_low_bits
```

Comment:
Huh, I can guess what `|=` does from context, but it's not what I would have expected that operator to do (maybe just because I'm not super familiar with bit operators). It just tacks on the low bits? 

Is this equivalent?
```
# bitshift left by 2 to make room for adding on the low-bits
data[:, byte::5] = (data[:, byte::5] << 2) + masked_low_bits
```

Reply:
Ooh, I didn't notice above that the `data` had already been bitshifted left in preparation. Nvm this makes sense now.

And your comment about switching to using `pixel_bytes_2d` finally makes sense too, as my follow-up confusion was about how that preparatory bit-shifting affected the fifth byte too.

Reply:
I just restructured this function to increase clarity. I did so without reading your last 2 comments first, and it turns out I implemented them both. high five ✋